File: client/src/pages/patient/PatientProfile.js
import React, { useEffect, useState } from 'react';
import toast from 'react-hot-toast';
import { patientAPI } from '../../utils/api';
import PhoneInput from '../../components/common/PhoneInput';
import LocationSelect from '../../components/common/LocationSelect';
import { clearDraft, loadDraft, saveDraft } from '../../utils/draftStorage';

const PATIENT_PROFILE_DRAFT_KEY = 'patient_profile_draft';
const PATIENT_INSURANCE_DRAFT_KEY = 'patient_insurance_draft';

export default function PatientProfile() {
  const [loading, setLoading] = useState(true);
  const [profile, setProfile] = useState(null);
  const [policies, setPolicies] = useState([]);
  const [insuranceForm, setInsuranceForm] = useState({
    provider: '',
    policy_number: '',
    plan_name: '',
    valid_from: '',
    valid_to: ''
  });
  const [insuranceFile, setInsuranceFile] = useState(null);
  const [form, setForm] = useState({
    date_of_birth: '',
    gender: '',
    blood_group: '',
    address: '',
    city: '',
    state: '',
    country: '',
    countryCode: '',
    stateCode: '',
    pincode: '',
    emergency_contact_name: '',
    emergency_contact_phone: '',
    allergies: '',
    chronic_conditions: '',
  });

  const load = async () => {
    setLoading(true);
    try {
      const { data } = await patientAPI.getProfile();
      const profileDraft = loadDraft(PATIENT_PROFILE_DRAFT_KEY);
      const nextForm = {
        date_of_birth: data.patient?.date_of_birth || '',
        gender: data.patient?.gender || '',
        blood_group: data.patient?.blood_group || '',
        address: data.patient?.address || '',
        city: data.patient?.city || '',
        state: data.patient?.state || '',
        country: data.patient?.country || '',
        countryCode: '',
        stateCode: '',
        pincode: data.patient?.pincode || '',
        emergency_contact_name: data.patient?.emergency_contact_name || '',
        emergency_contact_phone: data.patient?.emergency_contact_phone || '',
        allergies: data.patient?.allergies || '',
        chronic_conditions: data.patient?.chronic_conditions || '',
      };
      setProfile(data.patient);
      setForm(profileDraft?.form ? { ...nextForm, ...profileDraft.form } : nextForm);
      try {
        const insuranceRes = await patientAPI.getInsurance();
        setPolicies(insuranceRes.data?.policies || []);
      } catch {
        setPolicies([]);
      }
    } catch {
      setProfile(null);
      setPolicies([]);
    } finally {
      setLoading(false);
    }
  };

  useEffect(() => { load(); }, []);
  useEffect(() => {
    const insuranceDraft = loadDraft(PATIENT_INSURANCE_DRAFT_KEY);
    if (insuranceDraft?.form) {
      setInsuranceForm((prev) => ({ ...prev, ...insuranceDraft.form }));
    }
  }, []);
  useEffect(() => {
    if (loading) return;
    saveDraft(PATIENT_PROFILE_DRAFT_KEY, { form });
  }, [form, loading]);
  useEffect(() => {
    saveDraft(PATIENT_INSURANCE_DRAFT_KEY, { form: insuranceForm });
  }, [insuranceForm]);

  const save = async () => {
    try {
      await patientAPI.updateProfile(form);
      clearDraft(PATIENT_PROFILE_DRAFT_KEY);
      toast.success('Profile updated');
      load();
    } catch {
      toast.error('Failed to update profile');
    }
  };

  const saveInsurance = async () => {
    if (!insuranceForm.provider || !insuranceForm.policy_number) {
      return toast.error('Provider and policy number are required');
    }
    const fd = new FormData();
    fd.append('provider', insuranceForm.provider);
    fd.append('policy_number', insuranceForm.policy_number);
    if (insuranceForm.plan_name) fd.append('plan_name', insuranceForm.plan_name);
    if (insuranceForm.valid_from) fd.append('valid_from', insuranceForm.valid_from);
    if (insuranceForm.valid_to) fd.append('valid_to', insuranceForm.valid_to);
    if (insuranceFile) fd.append('document', insuranceFile);
    try {
      await patientAPI.addInsurance(fd);
      toast.success('Insurance saved');
      clearDraft(PATIENT_INSURANCE_DRAFT_KEY);
      setInsuranceForm({ provider: '', policy_number: '', plan_name: '', valid_from: '', valid_to: '' });
      setInsuranceFile(null);
      load();
    } catch {
      toast.error('Failed to save insurance');
    }
  };

  return (
    <div>
      <div className="card" style={{ marginBottom: 20 }}>
        <div className="card-header">
          <div className="card-title">Patient Profile</div>
        </div>
        <div className="card-body">
          {loading ? (
            <div style={{ color: 'var(--text-muted)' }}>Loading...</div>
          ) : profile ? (
            <div style={{ fontSize: 13, color: 'var(--text-muted)' }}>
              <div><strong>Name:</strong> {profile.name}</div>
              <div><strong>Phone:</strong> {profile.phone || '-'}</div>
              <div><strong>Email:</strong> {profile.email || '-'}</div>
            </div>
          ) : (
            <div style={{ color: 'var(--text-muted)' }}>Profile not found.</div>
          )}
        </div>
      </div>

      <div className="card">
        <div className="card-header">
          <div className="card-title">Update Details</div>
        </div>
        <div className="card-body">
          <div className="form-hint" style={{ marginBottom: 12 }}>
            Profile details are saved locally on this device while you work offline. File uploads still need internet.
          </div>
          <div className="form-grid">
            <div className="form-group">
              <label className="form-label">Date of Birth</label>
              <input className="form-input" type="date" value={form.date_of_birth} onChange={(e) => setForm({ ...form, date_of_birth: e.target.value })} />
            </div>
            <div className="form-group">
              <label className="form-label">Gender</label>
              <select className="form-select" value={form.gender} onChange={(e) => setForm({ ...form, gender: e.target.value })}>
                <option value="">Select</option>
                <option value="male">Male</option>
                <option value="female">Female</option>
                <option value="other">Other</option>
              </select>
            </div>
          </div>
          <div className="form-group">
            <label className="form-label">Blood Group</label>
            <select className="form-select" value={form.blood_group} onChange={(e) => setForm({ ...form, blood_group: e.target.value })}>
              <option value="">Select</option>
              {['A+','A-','B+','B-','AB+','AB-','O+','O-'].map((bg) => (
                <option key={bg} value={bg}>{bg}</option>
              ))}
            </select>
          </div>
          <div className="form-group">
            <label className="form-label">Address</label>
            <input className="form-input" value={form.address} onChange={(e) => setForm({ ...form, address: e.target.value })} />
          </div>
          <LocationSelect
            value={{
              country: form.country,
              countryCode: form.countryCode,
              state: form.state,
              stateCode: form.stateCode,
              city: form.city,
            }}
            onChange={(loc) =>
              setForm((prev) => ({
                ...prev,
                country: loc.country,
                countryCode: loc.countryCode,
                state: loc.state,
                stateCode: loc.stateCode,
                city: loc.city,
              }))
            }
          />
          <div className="form-grid">
            <div className="form-group">
              <label className="form-label">Pincode</label>
              <input className="form-input" value={form.pincode} onChange={(e) => setForm({ ...form, pincode: e.target.value })} />
            </div>
            <div className="form-group">
              <label className="form-label">Emergency Contact Name</label>
              <input className="form-input" value={form.emergency_contact_name} onChange={(e) => setForm({ ...form, emergency_contact_name: e.target.value })} />
            </div>
          </div>
          <div className="form-group">
            <label className="form-label">Emergency Contact Phone</label>
            <PhoneInput value={form.emergency_contact_phone} onChange={(val) => setForm({ ...form, emergency_contact_phone: val })} />
          </div>
          <div className="form-group">
            <label className="form-label">Allergies</label>
            <input className="form-input" value={form.allergies} onChange={(e) => setForm({ ...form, allergies: e.target.value })} />
          </div>
          <div className="form-group">
            <label className="form-label">Chronic Conditions</label>
            <input className="form-input" value={form.chronic_conditions} onChange={(e) => setForm({ ...form, chronic_conditions: e.target.value })} />
          </div>
          <button className="btn btn-primary" onClick={save}>Save Profile</button>
        </div>
      </div>

      <div className="card" style={{ marginTop: 20 }}>
        <div className="card-header">
          <div className="card-title">Insurance & Policy (KYC)</div>
        </div>
        <div className="card-body">
          <div className="form-hint" style={{ marginBottom: 12 }}>
            Insurance form details are saved locally. If you lose internet, you may need to attach the KYC document again after reconnecting.
          </div>
          {policies.length === 0 ? (
            <div style={{ color: 'var(--text-muted)', marginBottom: 12 }}>No insurance policies added yet.</div>
          ) : (
            <div className="table-container" style={{ marginBottom: 12 }}>
              <table>
                <thead>
                  <tr>
                    <th>Provider</th>
                    <th>Policy #</th>
                    <th>Status</th>
                    <th>Valid</th>
                  </tr>
                </thead>
                <tbody>
                  {policies.map((p) => (
                    <tr key={p.id}>
                      <td>{p.provider}</td>
                      <td>{p.policy_number}</td>
                      <td>{p.status || 'pending'}</td>
                      <td>{p.valid_from || '-'} {p.valid_to ? `- ${p.valid_to}` : ''}</td>
                    </tr>
                  ))}
                </tbody>
              </table>
            </div>
          )}

          <div className="form-grid">
            <div className="form-group">
              <label className="form-label">Provider<span className="required">*</span></label>
              <input className="form-input" value={insuranceForm.provider} onChange={(e) => setInsuranceForm({ ...insuranceForm, provider: e.target.value })} />
            </div>
            <div className="form-group">
              <label className="form-label">Policy Number<span className="required">*</span></label>
              <input className="form-input" value={insuranceForm.policy_number} onChange={(e) => setInsuranceForm({ ...insuranceForm, policy_number: e.target.value })} />
            </div>
          </div>
          <div className="form-group">
            <label className="form-label">Plan Name</label>
            <input className="form-input" value={insuranceForm.plan_name} onChange={(e) => setInsuranceForm({ ...insuranceForm, plan_name: e.target.value })} />
          </div>
          <div className="form-grid">
            <div className="form-group">
              <label className="form-label">Valid From</label>
              <input className="form-input" type="date" value={insuranceForm.valid_from} onChange={(e) => setInsuranceForm({ ...insuranceForm, valid_from: e.target.value })} />
            </div>
            <div className="form-group">
              <label className="form-label">Valid To</label>
              <input className="form-input" type="date" value={insuranceForm.valid_to} onChange={(e) => setInsuranceForm({ ...insuranceForm, valid_to: e.target.value })} />
            </div>
          </div>
          <div className="form-group">
            <label className="form-label">KYC Document</label>
            <input className="form-input" type="file" onChange={(e) => setInsuranceFile(e.target.files?.[0] || null)} />
          </div>
          <button className="btn btn-primary" onClick={saveInsurance}>Save Insurance</button>
        </div>
      </div>
    </div>
  );
}

File: mobile/medicare_pro_app/lib/features/patient/patient_profile.dart
import 'package:csc_picker/csc_picker.dart';
import 'package:file_picker/file_picker.dart';
import 'package:flutter/material.dart';

import '../../core/api/api_client.dart';
import '../../core/services/draft_store.dart';

const _patientProfileDraftKey = 'patient_profile_mobile_draft';
const _patientInsuranceDraftKey = 'patient_insurance_mobile_draft';

class PatientProfilePage extends StatefulWidget {
  const PatientProfilePage({super.key});

  @override
  State<PatientProfilePage> createState() => _PatientProfilePageState();
}

class _PatientProfilePageState extends State<PatientProfilePage> {
  Map<String, dynamic>? _profile;
  bool _loading = false;
  bool _hydratingDraft = false;

  final _nameCtrl = TextEditingController();
  final _dobCtrl = TextEditingController();
  final _genderCtrl = TextEditingController();
  final _bloodCtrl = TextEditingController();
  final _addressCtrl = TextEditingController();
  final _pincodeCtrl = TextEditingController();
  final _emgNameCtrl = TextEditingController();
  final _emgPhoneCtrl = TextEditingController();
  final _allergiesCtrl = TextEditingController();
  final _chronicCtrl = TextEditingController();

  List<dynamic> _policies = [];
  final _providerCtrl = TextEditingController();
  final _policyNumberCtrl = TextEditingController();
  final _planCtrl = TextEditingController();
  final _validFromCtrl = TextEditingController();
  final _validToCtrl = TextEditingController();
  String? _docPath;
  String? _country;
  String? _state;
  String? _city;

  Future<void> _saveProfileDraft() async {
    if (_hydratingDraft) return;
    await DraftStore.save(_patientProfileDraftKey, {
      'name': _nameCtrl.text.trim(),
      'date_of_birth': _dobCtrl.text.trim(),
      'gender': _genderCtrl.text.trim(),
      'blood_group': _bloodCtrl.text.trim(),
      'address': _addressCtrl.text.trim(),
      'country': _country,
      'state': _state,
      'city': _city,
      'pincode': _pincodeCtrl.text.trim(),
      'emergency_contact_name': _emgNameCtrl.text.trim(),
      'emergency_contact_phone': _emgPhoneCtrl.text.trim(),
      'allergies': _allergiesCtrl.text.trim(),
      'chronic_conditions': _chronicCtrl.text.trim(),
    });
  }

  Future<void> _saveInsuranceDraft() async {
    if (_hydratingDraft) return;
    await DraftStore.save(_patientInsuranceDraftKey, {
      'provider': _providerCtrl.text.trim(),
      'policy_number': _policyNumberCtrl.text.trim(),
      'plan_name': _planCtrl.text.trim(),
      'valid_from': _validFromCtrl.text.trim(),
      'valid_to': _validToCtrl.text.trim(),
    });
  }

  void _bindDraftListeners() {
    final profileListeners = [
      _nameCtrl,
      _dobCtrl,
      _genderCtrl,
      _bloodCtrl,
      _addressCtrl,
      _pincodeCtrl,
      _emgNameCtrl,
      _emgPhoneCtrl,
      _allergiesCtrl,
      _chronicCtrl,
    ];
    for (final controller in profileListeners) {
      controller.addListener(_saveProfileDraft);
    }

    final insuranceListeners = [
      _providerCtrl,
      _policyNumberCtrl,
      _planCtrl,
      _validFromCtrl,
      _validToCtrl,
    ];
    for (final controller in insuranceListeners) {
      controller.addListener(_saveInsuranceDraft);
    }
  }

  Future<void> _load() async {
    setState(() => _loading = true);
    try {
      final profileDraft = await DraftStore.load(_patientProfileDraftKey);
      final insuranceDraft = await DraftStore.load(_patientInsuranceDraftKey);
      final data =
          await ApiClient.instance.get('/api/patient/profile', auth: true);
      final profile = (data['profile'] as Map?)?.cast<String, dynamic>();

      _hydratingDraft = true;
      setState(() => _profile = profile);
      if (profile != null) {
        _nameCtrl.text = profile['name'] ?? '';
        _dobCtrl.text = profile['date_of_birth'] ?? '';
        _genderCtrl.text = profile['gender'] ?? '';
        _bloodCtrl.text = profile['blood_group'] ?? '';
        _addressCtrl.text = profile['address'] ?? '';
        _city = profile['city'];
        _state = profile['state'];
        _country = profile['country'];
        _pincodeCtrl.text = profile['pincode'] ?? '';
        _emgNameCtrl.text = profile['emergency_contact_name'] ?? '';
        _emgPhoneCtrl.text = profile['emergency_contact_phone'] ?? '';
        _allergiesCtrl.text = profile['allergies'] ?? '';
        _chronicCtrl.text = profile['chronic_conditions'] ?? '';
      }

      if (profileDraft != null) {
        _nameCtrl.text = profileDraft['name']?.toString() ?? _nameCtrl.text;
        _dobCtrl.text =
            profileDraft['date_of_birth']?.toString() ?? _dobCtrl.text;
        _genderCtrl.text = profileDraft['gender']?.toString() ?? _genderCtrl.text;
        _bloodCtrl.text =
            profileDraft['blood_group']?.toString() ?? _bloodCtrl.text;
        _addressCtrl.text =
            profileDraft['address']?.toString() ?? _addressCtrl.text;
        _country = profileDraft['country']?.toString() ?? _country;
        _state = profileDraft['state']?.toString() ?? _state;
        _city = profileDraft['city']?.toString() ?? _city;
        _pincodeCtrl.text =
            profileDraft['pincode']?.toString() ?? _pincodeCtrl.text;
        _emgNameCtrl.text = profileDraft['emergency_contact_name']?.toString() ??
            _emgNameCtrl.text;
        _emgPhoneCtrl.text =
            profileDraft['emergency_contact_phone']?.toString() ??
                _emgPhoneCtrl.text;
        _allergiesCtrl.text =
            profileDraft['allergies']?.toString() ?? _allergiesCtrl.text;
        _chronicCtrl.text = profileDraft['chronic_conditions']?.toString() ??
            _chronicCtrl.text;
      }

      if (insuranceDraft != null) {
        _providerCtrl.text =
            insuranceDraft['provider']?.toString() ?? _providerCtrl.text;
        _policyNumberCtrl.text = insuranceDraft['policy_number']?.toString() ??
            _policyNumberCtrl.text;
        _planCtrl.text =
            insuranceDraft['plan_name']?.toString() ?? _planCtrl.text;
        _validFromCtrl.text =
            insuranceDraft['valid_from']?.toString() ?? _validFromCtrl.text;
        _validToCtrl.text =
            insuranceDraft['valid_to']?.toString() ?? _validToCtrl.text;
      }
      _hydratingDraft = false;

      await _loadPolicies();
    } catch (e) {
      if (!mounted) return;
      ScaffoldMessenger.of(context)
          .showSnackBar(SnackBar(content: Text('Load failed: $e')));
    } finally {
      _hydratingDraft = false;
      if (mounted) setState(() => _loading = false);
    }
  }

  Future<void> _loadPolicies() async {
    try {
      final data =
          await ApiClient.instance.get('/api/patient/insurance', auth: true);
      final list = data['policies'] as List? ?? [];
      if (mounted) setState(() => _policies = list);
    } catch (_) {
      if (mounted) setState(() => _policies = []);
    }
  }

  Future<void> _pickDoc() async {
    final res = await FilePicker.platform.pickFiles();
    if (res != null && res.files.isNotEmpty) {
      setState(() => _docPath = res.files.single.path);
    }
  }

  Future<void> _saveInsurance() async {
    try {
      await ApiClient.instance.postMultipart(
        '/api/patient/insurance',
        {
          'provider': _providerCtrl.text.trim(),
          'policy_number': _policyNumberCtrl.text.trim(),
          'plan_name': _planCtrl.text.trim(),
          'valid_from': _validFromCtrl.text.trim(),
          'valid_to': _validToCtrl.text.trim(),
        },
        filePath: _docPath,
        auth: true,
      );
      await DraftStore.clear(_patientInsuranceDraftKey);
      if (!mounted) return;
      ScaffoldMessenger.of(context)
          .showSnackBar(const SnackBar(content: Text('Insurance saved')));
      _providerCtrl.clear();
      _policyNumberCtrl.clear();
      _planCtrl.clear();
      _validFromCtrl.clear();
      _validToCtrl.clear();
      setState(() => _docPath = null);
      _loadPolicies();
    } catch (e) {
      if (!mounted) return;
      ScaffoldMessenger.of(context)
          .showSnackBar(SnackBar(content: Text('Save failed: $e')));
    }
  }

  Future<void> _save() async {
    try {
      final res = await ApiClient.instance.post('/api/patient/profile', {
        'name': _nameCtrl.text.trim(),
        'date_of_birth': _dobCtrl.text.trim(),
        'gender': _genderCtrl.text.trim(),
        'blood_group': _bloodCtrl.text.trim(),
        'address': _addressCtrl.text.trim(),
        'country': _country ?? '',
        'city': _city ?? '',
        'state': _state ?? '',
        'pincode': _pincodeCtrl.text.trim(),
        'emergency_contact_name': _emgNameCtrl.text.trim(),
        'emergency_contact_phone': _emgPhoneCtrl.text.trim(),
        'allergies': _allergiesCtrl.text.trim(),
        'chronic_conditions': _chronicCtrl.text.trim(),
      }, auth: true);
      await DraftStore.clear(_patientProfileDraftKey);
      if (!mounted) return;
      ScaffoldMessenger.of(context).showSnackBar(
        SnackBar(
          content: Text(
            res['offlineQueued'] == true
                ? 'Profile saved offline and queued for sync.'
                : 'Profile updated',
          ),
        ),
      );
    } catch (e) {
      if (!mounted) return;
      ScaffoldMessenger.of(context)
          .showSnackBar(SnackBar(content: Text('Save failed: $e')));
    }
  }

  @override
  void initState() {
    super.initState();
    _bindDraftListeners();
    _load();
  }

  @override
  void dispose() {
    _nameCtrl.dispose();
    _dobCtrl.dispose();
    _genderCtrl.dispose();
    _bloodCtrl.dispose();
    _addressCtrl.dispose();
    _pincodeCtrl.dispose();
    _emgNameCtrl.dispose();
    _emgPhoneCtrl.dispose();
    _allergiesCtrl.dispose();
    _chronicCtrl.dispose();
    _providerCtrl.dispose();
    _policyNumberCtrl.dispose();
    _planCtrl.dispose();
    _validFromCtrl.dispose();
    _validToCtrl.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('My Profile')),
      body: _loading
          ? const Center(child: CircularProgressIndicator())
          : ListView(
              padding: const EdgeInsets.all(16),
              children: [
                const Text(
                  'Profile details stay saved on this device while offline and sync when internet returns.',
                ),
                const SizedBox(height: 12),
                TextField(
                  controller: _nameCtrl,
                  decoration: const InputDecoration(labelText: 'Full Name'),
                ),
                TextField(
                  decoration: InputDecoration(
                    labelText: 'Email (read-only)',
                    hintText: _profile?['email'] ?? '',
                  ),
                  enabled: false,
                ),
                TextField(
                  decoration: InputDecoration(
                    labelText: 'Phone (read-only)',
                    hintText: _profile?['phone'] ?? '',
                  ),
                  enabled: false,
                ),
                const SizedBox(height: 8),
                TextField(
                  controller: _dobCtrl,
                  decoration: const InputDecoration(
                    labelText: 'Date of Birth (YYYY-MM-DD)',
                  ),
                ),
                TextField(
                  controller: _genderCtrl,
                  decoration: const InputDecoration(labelText: 'Gender'),
                ),
                TextField(
                  controller: _bloodCtrl,
                  decoration: const InputDecoration(labelText: 'Blood Group'),
                ),
                TextField(
                  controller: _addressCtrl,
                  decoration: const InputDecoration(labelText: 'Address'),
                ),
                const SizedBox(height: 8),
                CSCPicker(
                  layout: Layout.vertical,
                  currentCountry: _country,
                  currentState: _state,
                  currentCity: _city,
                  countryDropdownLabel: 'Select Country',
                  stateDropdownLabel: 'Select State',
                  cityDropdownLabel: 'Select City',
                  onCountryChanged: (value) {
                    setState(() {
                      _country = value;
                      _state = null;
                      _city = null;
                    });
                    _saveProfileDraft();
                  },
                  onStateChanged: (value) {
                    setState(() {
                      _state = value;
                      _city = null;
                    });
                    _saveProfileDraft();
                  },
                  onCityChanged: (value) {
                    setState(() => _city = value);
                    _saveProfileDraft();
                  },
                ),
                TextField(
                  controller: _pincodeCtrl,
                  decoration: const InputDecoration(labelText: 'Pincode'),
                ),
                const SizedBox(height: 8),
                TextField(
                  controller: _emgNameCtrl,
                  decoration: const InputDecoration(
                    labelText: 'Emergency Contact Name',
                  ),
                ),
                TextField(
                  controller: _emgPhoneCtrl,
                  decoration: const InputDecoration(
                    labelText: 'Emergency Contact Phone',
                  ),
                ),
                TextField(
                  controller: _allergiesCtrl,
                  decoration: const InputDecoration(labelText: 'Allergies'),
                ),
                TextField(
                  controller: _chronicCtrl,
                  decoration:
                      const InputDecoration(labelText: 'Chronic Conditions'),
                ),
                const SizedBox(height: 12),
                ElevatedButton(
                  onPressed: _save,
                  child: const Text('Save Profile'),
                ),
                const SizedBox(height: 16),
                const Text(
                  'Insurance & Policy (KYC)',
                  style: TextStyle(fontSize: 16, fontWeight: FontWeight.w700),
                ),
                const SizedBox(height: 8),
                const Text(
                  'Insurance form details stay saved locally. KYC document upload still needs internet and may need to be reattached.',
                ),
                const SizedBox(height: 8),
                if (_policies.isEmpty)
                  const Text('No insurance policies added yet')
                else
                  ..._policies.map((policy) {
                    final provider = policy['provider'] ?? '';
                    final policyNumber = policy['policy_number'] ?? '';
                    final status = policy['status'] ?? 'pending';
                    return ListTile(
                      contentPadding: EdgeInsets.zero,
                      title: Text(provider),
                      subtitle: Text('Policy: $policyNumber'),
                      trailing: Text(status.toString()),
                    );
                  }).toList(),
                const Divider(),
                TextField(
                  controller: _providerCtrl,
                  decoration: const InputDecoration(labelText: 'Provider *'),
                ),
                TextField(
                  controller: _policyNumberCtrl,
                  decoration:
                      const InputDecoration(labelText: 'Policy Number *'),
                ),
                TextField(
                  controller: _planCtrl,
                  decoration: const InputDecoration(labelText: 'Plan Name'),
                ),
                TextField(
                  controller: _validFromCtrl,
                  decoration: const InputDecoration(
                    labelText: 'Valid From (YYYY-MM-DD)',
                  ),
                ),
                TextField(
                  controller: _validToCtrl,
                  decoration:
                      const InputDecoration(labelText: 'Valid To (YYYY-MM-DD)'),
                ),
                const SizedBox(height: 8),
                OutlinedButton(
                  onPressed: _pickDoc,
                  child: Text(
                    _docPath == null
                        ? 'Upload KYC Document'
                        : 'Document Selected',
                  ),
                ),
                const SizedBox(height: 8),
                ElevatedButton(
                  onPressed: _saveInsurance,
                  child: const Text('Save Insurance'),
                ),
              ],
            ),
    );
  }
}
